rotkehlchen/tests/unit/test_cost_basis.py

```diff
@@ -580,3 +584,59 @@ def test_missing_acquisitions(accountant):
         time=4,
     ))
     assert cost_basis.missing_acquisitions == expected_missing_acquisitions
+
+
+@pytest.mark.parametrize('mocked_price_queries', [{
+    A_ETH: {A_EUR: {1469020840: ONE}},
+    A_3CRV: {A_EUR: {1469020840: ONE}},
+}])
+def test_swaps_acquisitions(accountant):
+    """Check acquisition part of swaps doesn't count as taxable."""
+    pot = accountant.pots[0]
+    transactions_accountant = pot.transactions
+    transactions_accountant._process_tx_swap(
+        timestamp=1469020840,
+        out_event=HistoryBaseEntry(
+            event_identifier=b'123',
+            sequence_index=1,
+            timestamp=Timestamp(1469020840),
+            location=Location.BLOCKCHAIN,
+            location_label='location-label',
+            asset=A_ETH,
+            balance=Balance(amount=ONE, usd_value=ONE),
+            notes='',
+            event_type=HistoryEventType.TRADE,
+            event_subtype=HistoryEventSubType.SPEND,
+            counterparty='counterparty',
+        ),
+        in_event=HistoryBaseEntry(
+            event_identifier=b'123',
+            sequence_index=2,
+            timestamp=Timestamp(1469020840),
+            location=Location.BLOCKCHAIN,
+            location_label='location-label',
+            asset=A_3CRV,
+            balance=Balance(amount=ONE, usd_value=ONE),
+            notes='',
+            event_type=HistoryEventType.TRADE,
+            event_subtype=HistoryEventSubType.RECEIVE,
+            counterparty='counterparty',
+        ),
+        event_settings=TxEventSettings(
+            taxable=True,
+            count_entire_amount_spend=False,
+            count_cost_basis_pnl=True,
+            method='spend',
+            take=2,
+            multitake_treatment=TxMultitakeTreatment.SWAP,
+        ),
+    )
+    assert len(pot.processed_events) == 2
+    assert pot.processed_events[0].taxable_amount == ONE
+    assert pot.processed_events[0].free_amount == ZERO
+    assert pot.processed_events[0].pnl.taxable == ONE
```

Comment:
It would make sense to add another test to test the taxable and free amount being different, as (if I understand correctly) there is a bug in the current implementation